Otel end user discussions february 2023 (#2456)
open-telemetry/opentelemetry.io: content/en/blog/2023/end-user-discussions-02.md

---
title: OpenTelemetry End-User Discussions Summary for February 2023
linkTitle: End-User Discussions Feb 2023

> date: 2023-02-27

Reviewer: ```suggestion
date: 2023-03-07
```

author: >-
  [Pranay Prateek](https://github.com/pranay01) (SigNoz)
---

_With contributions from [Henrik Rexed](https://github.com/henrikrexed)
(Dynatrace), [Michael Hausenblas](https://github.com/mhausenblas) (AWS),
[Rynn Mancuso](https://github.com/musingvirtual) (Honeycomb),
[Reese Lee](https://github.com/reese-lee) (New Relic) and

> [Adriana Villela](https://github.com/avillela) (Lightstep)_

Reviewer: ```suggestion
[Adriana Villela](https://github.com/avillela) (Lightstep)._
{.small}
```

The discussions take place using a
[Lean Coffee format](https://agilecoffee.com/leancoffee/), whereby folks are
invited to post their topics to the
[Agile Coffee board like this one](http://agile.coffee/#3716060f-183a-4966-8da4-60daab2842c4),
and everyone in attendance votes on what they want to talk about.

## What we talked about

Some interesting topics that were discussed this month were:

- Sampling for traces
- Emitting business metrics
- Monitoring the health of OpenTelemetry Collector
- Backup/Buffer capabilities of OTel Collector

## Discussion Highlights

Below is the summary of this month's discussions.

### OpenTelemetry Collector

#### 1 - Monitoring OTel Collector's health

**Q:** Are there any suggestions for monitoring OTel Collector's health or
patterns for collecting agent telemetry?

**A:** Collectors can be used to collect telemetry data from other Collectors,
which doesn't really need to be a disparate telemetry system. Users should also
think about collecting multiple signals so that even if one signal fails, they
get alerted by another. Here's an
[article](https://ref.otel.help/otel-collector-ops/) discussing this.

> #### 2 - Timeline for opamp extension

Reviewer: ```suggestion
#### 2 - Timeline for OpAMP extension
```

**Q:** Is there any timeline for implementing the

> [opamp spec](https://github.com/open-telemetry/opamp-spec) for agent management?

Reviewer: ```suggestion
[OpAMP spec](https://github.com/open-telemetry/opamp-spec) for agent management?
```

**A:** It's not a top priority as of now. It would be good to have a maintainer

> from the community for opamp. You can track the issue
> [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16462).

Reviewer: For future reference, avoid having links with link text "here". (I'll fix one, but leave the rest for now.)

```suggestion
from the community for OpAMP. To track progress, see 
[issue #16462](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16462).
```

#### 3 - Buffer capabilities of OTel Collector

**Q:** What are some backup/retry buffer capabilities of OTel Collector when
endpoints are unavailable?

**A:** There is an experimental
[storage extension](https://github.com/open-telemetry/opentelemetry-collector/tree/main/extension/experimental/storage)
that is currently under development to support buffering and data persistence.

#### 4 - Periodically profiling Collectors to improve performance

**Q:** Is there any effort to periodically profile the Collector and improve
performance on an ongoing basis?

**A:** There is a GitHub action that runs load test on OpenTelemetry Collector,
but noone is working to improve it.

### OpenTelemetry Language API & SDKs

#### 1 - Timeline for Go SDK

**Q:** What is the timeline for full specification compliance for OTel Go SDK?

**A:** In the Go OTel SDK, the current progress is mostly around metrics. The

> logging development is frozen. Major work is being done in metrics sdk. Go
> metrics progress can be tracked
> [here](https://github.com/open-telemetry/opentelemetry-go/projects?query=is%3Aopen).

Reviewer: ```suggestion
logging development is frozen. Major work is being done in metrics sdk. To track
progress on Go metrics, see the
[Metric project tables](https://github.com/open-telemetry/opentelemetry-go/projects?query=metric).
```

which can instantiate other receivers at runtime based on whether an observed
endpoint matches a configured rule. To use the receiver creator, you must first
configure one or more
[observers](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/observer/README.md).
Using
[Kubernetes observer](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/k8sobserver#kubernetes-observer),
users should be able to detect and report Kubernetes pod, port, and node
endpoints via the Kubernetes API.

#### Hosting pattern suggestion of the OTel Collector within Azure

**Q:** Are there any suggestions for hosting pattern of the Collector within
Azure to collect telemetry from Azure App Services and Azure functions?

**A:** Usually, the community relies on folks from Microsoft to provide best
practices. There is some issue with the latest version of OTel and Azure
functions. You can track it
[here](https://github.com/Azure/azure-functions-host/issues/8938).

## Meeting Notes & Recordings

For a deeper dive into the above topics, check out the following:

> - [AMER](https://docs.google.com/document/d/1p_FoGbLiDC9VPqqLblJqQtHBn3tr-aPxhu2GaIykU6k/edit?usp=sharing)
>   meeting notes
> - [EMEA](https://docs.google.com/document/d/1fh4RWyZ-ScWdwrgpRHO9mnfqLSKfxUTf4wZGdUvnnUM/edit?usp=sharing)
>   meeting notes
> - [APAC](https://docs.google.com/document/d/1eDYC97LfvE428cpIf3A_hSGirdNzglPurlxgKCmw8o4/edit?usp=sharing)
>   meeting notes

Reviewer: ```suggestion
- [AMER](https://docs.google.com/document/d/1p_FoGbLiDC9VPqqLblJqQtHBn3tr-aPxhu2GaIykU6k)
  meeting notes
- [EMEA](https://docs.google.com/document/d/1fh4RWyZ-ScWdwrgpRHO9mnfqLSKfxUTf4wZGdUvnnUM)
  meeting notes
- [APAC](https://docs.google.com/document/d/1eDYC97LfvE428cpIf3A_hSGirdNzglPurlxgKCmw8o4)
  meeting notes
```
